packages/amplify-sdk/src/amplify-sdk.js

```diff
@@ -215,13 +243,37 @@ export default class AmplifySDK {
 					}
 				}
 
+				// do the login
 				account = await this.client.login(opts);
 
+				// if we're in manual mode (e.g. --no-launch-browser), then return now
 				if (opts.manual) {
 					// account is actually an object containing `cancel`, `promise`, and `url`
 					return account;
 				}
 
+				// upgrade the service account with the platform tooling account
+				if (username && password) {
+					try {
+						await this.request('/api/v1/auth/login', account, {
+							errorMsg: 'Failed to authenticate',
+							isToolingAuth: true,
+							json: {
+								from: 'cli',
```

Comment:
what does this from: 'cli' key/value pair do?

Reply:
I can't be certain what it does. It's a magical property.